libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,452 @@
+  private _clientRect$: Observable<ClientRect>;
+  /** Observer that gets triggered if the slider is resized on the screen. */
+  private _resizeObserver$ = new Subject<void>();
+  /** Variable to hold the ResizeObserver */
+  private _observer: any;
+  /** Observer that gets triggered if the input field value is changed. */
+  private _inputFieldValue$ = new Subject<number>();
+
+  /**
+   * Binding for the minimum value of the slider.
+   */
+  @Input()
+  get min(): number {
+    return this._min;
+  }
+  set min(min: number) {
+    this._min = coerceNumberProperty(min);
+    if (this._min > this._value) {
+      this._updateValue(this._min, true);
+    } else {
+      this._updateSliderPosition(this._value, this._min, this._max);
+    }
+  }
+  private _min: number = 0;
+
+  /**
+   * Binding for the maximum value of the slider.
+   */
+  @Input()
+  get max(): number {
+    return this._max;
+  }
+  set max(max: number) {
+    this._max = coerceNumberProperty(max);
+    if (this._max < this._value) {
+      this._updateValue(this._max, true);
+    } else {
+      this._updateSliderPosition(this._value, this._min, this._max);
+    }
+  }
+  private _max: number = 10;
+
+  /**
+   * Bindings for the step, if changed, roundShift needs to be recalculated.
+   */
+  @Input()
+  get step(): number {
+    return this._step;
+  }
+  set step(step: number) {
+    if (isDefined(step)) {
+      this._step = coerceNumberProperty(step);
+    }
+
+    this._roundShift = 0;
+    // deal with rounding problems.
+    let countingStep = this._step;
+
+    while (countingStep % 1 !== 0) {
+      countingStep *= 10;
+      this._roundShift++;
+    }
+  }
+  /** Holds the value of step internally */
+  private _step: number = 1;
+
+  /** Binding for the disabled state. */
+  @Input()
+  get disabled(): boolean {
+    return this._isDisabled;
+  }
+  set disabled(disabled: boolean) {
+    this._isDisabled = coerceBooleanProperty(disabled);
+    this._changeDetectionRef.markForCheck();
+  }
+  private _isDisabled: boolean = false;
+
+  /**
+   * The binding for the value of the slider.
+   */
+  @Input()
+  get value(): number {
+    return this._value;
+  }
+  set value(value: number) {
+    this._updateValue(coerceNumberProperty(value));
+  }
+  private _value: number = 0;
+
+  /** Updates the value if the update is triggered by the consumer. */
+  private _updateValue(value: number, userTriggered: boolean = true): void {
+    this._value = value;
+    // We only need to update if the update is coming from outside the component.
+    if (userTriggered) {
+      this._value$.next(roundToSnap(value, this.step, this._min, this._max));
+    }
+  }
+
+  /** Observer that completes on ngOnDestroy */
+  private _destroy$ = new Subject<void>();
+
+  /**
+   * Convert input string value to number and call
+   * roundToSnap takes care of snapping the values to the steps
+   */
+  inputValueChanged(event: Event): void {
```

Comment:
I think this method doesn't need to be part of the public API please make internal.

Reply:
I'll make it internal.